### PR TITLE
Merge method-level and class-level @ApiResponses annotations.

### PR DESCRIPTION
**Pull request by sworisbreathing**
#### What's this PR do/fix?
Merges `@ApiResponses` annotations from method-level and type-level (at least, when annotated in the same class). Method-level annotations take priority over type-level annotations.
#### Are there unit tests? If not how should this be manually tested?
I have updated `AnnotationsSpec` to verify the behavior when the annotations are defined in the same class. I have not tested subclassing behavior. I have also manually tested against the example project I created for #1465 (https://github.com/sworisbreathing/springfox-issues)
#### Any background context you want to provide?
See #1465.
#### What are the relevant issues?
#1465

### CODE REVIEW

File: springfox-swagger-common/src/main/java/springfox/documentation/swagger/annotations/Annotations.java [hunk 1]
```diff
@@ -52,8 +53,16 @@ private Annotations() {
     return fromNullable(findAnnotation(annotated, ApiOperation.class));
   }
 
-  public static Optional<ApiResponses> findApiResponsesAnnotations(Method annotated) {
-    return fromNullable(findAnnotation(annotated, ApiResponses.class));
+  public static List<Optional<ApiResponses>> findApiResponsesAnnotations(AnnotatedElement annotated) {
```
Comment (dilipkrish) [inline springfox-swagger-common/src/main/java/springfox/documentation/swagger/annotations/Annotations.java]: `List` is monadic in itself, so just a returning a list of ApiResponses would suffice. i.e. don't add it to the list of the `getAnnotation` method returns a null. makes sense? 

It'll make consumption in downstream processes easier, for e.g. we won't need a `if (apiResponses.isPresent()) {...}` block.

**Comment by dilipkrish** [comment]
👍 looks good except for a minor nit! thanks for the PR! Super appreciated!

**Comment by sworisbreathing** [comment]
Okay I've updated the PR to get rid of the `Optional`s in the list. Let me know what you think!

**Comment by dilipkrish** [comment]
Thanks it looks great! Its in!! :bow:

**Pull request merged**